### ISSUE TITLE
UVFITS rdate question

### ISSUE THREAD

**Issue by telegraphic** [RadioAstronomySoftwareGroup/pyuvdata#1433]
Hi all, wondering whether [this UVFITS code](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/4fba712c51d638ed12b6669b157c2926b152b2a0/pyuvdata/uvdata/uvfits.py#L1342): 

```python
        if (self.rdate is None) or (self.rdate == ""):
            rdate_obj = Time(np.floor(self.time_array[0]), format="jd", scale="utc")
```

Should account for the 0.5 Julian Date fraction offset due to UT1? E.g. 

```python
        if (self.rdate is None) or (self.rdate == ""):
            rdate_obj = Time(np.floor(self.time_array[0]) + 0.5, format="jd", scale="utc")
```

**Comment by kartographer** [comment]
Hey @telegraphic! The astropy `Time` object handles (or at least, _should_ handle) the JD "noon offset" under the hood when converting to Gregorian dates. But I did do a quick check to make sure that was the case:
```
In [1]: from astropy.time import Time

In [2]: Time(2456789.0, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[2]: '2014-05-11'

In [3]: Time(2456789.49, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[3]: '2014-05-11'

In [4]: Time(2456789.5, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[4]: '2014-05-12'

In [5]: Time(2456789.51, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[5]: '2014-05-12'
```
Looks like it behaves as expected to me! 😄

**Comment by telegraphic** [comment]
@kartographer  agreed, but notice there's an errant `np.floor` in the code:

```
In [4]: Time(np.floor(2456789.5), format='jd',scale='utc').strftime("%Y-%m-%d")
Out[4]: '2014-05-11'

In [4]: Time(2456789.5, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[4]: '2014-05-12'

In [5]: Time(np.floor(2456789.51), format='jd',scale='utc').strftime("%Y-%m-%d")
Out[5]: '2014-05-11'

In [5]: Time(2456789.51, format='jd',scale='utc').strftime("%Y-%m-%d")
Out[5]: '2014-05-12'
```

So I think my suggestion is: remove `np.floor`?

I have now also noticed that `DATE-OBS` in the AIPS 117 memo is YYYY-MM-DD, and pyuvdata is currently writing the full ISOT:

```
        hdu.header["DATE-OBS"] = Time(self.time_array[0], scale="utc", format="jd").isot
```
which I think should be changed to:
```
        hdu.header["DATE-OBS"] = Time(self.time_array[0], scale="utc", format="jd").strftime("%Y-%m-%d")
```

I can confirm that MIRIAD is confused when importing UVFITS files, as it uses both DATE-OBS and RDATE

**Comment by kartographer** [comment]
Ahh, I totally missed the call to `np.floor` -- yes, _that's_ definitely a bug that should be fixed.

**Comment by telegraphic** [comment]
Thanks @kartographer , I wasn't sure if there was a motivation for it, as it seemed rather deliberate! I've issued a PR from my branch. The PR #1436 did introduce a unit test failure with DUT1 now slightly different in one test -- not sure if the new value or golden value is off.

**Comment by kartographer** [comment]
Closed in #1439
